### PR TITLE
Converter + networks settings + networks group policies

### PR DESCRIPTION
**Pull request by mcparaf**
This pull request includes a few things:

1. The converter binary - a program which, provided an OpenAPI specification, can generate almost complete input files for the provider code generator
2. The networks settings resource generated using the converter
3. The networks group policy resource generated using the converter

Unfortunately, the generated yaml files cannot be completely automatically fed to the code generator, because of incompleteness of OpenAPI specs, Meraki quirks and details related to our testing environment. It does provide a good baseline though.

To learn how to use the converter please see the usage string.

### CODE REVIEW

File: gen/templates/data_source.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ func (d *{{camelCase .Name}}DataSource) Schema(ctx context.Context, req datasour
 func (d *{{camelCase .Name}}DataSource) ConfigValidators(ctx context.Context) []datasource.ConfigValidator {
     return []datasource.ConfigValidator{
         datasourcevalidator.ExactlyOneOf(
-            path.MatchRoot("id"),
+            path.MatchRoot("id"), // is this correct?
```
Comment (danischm) [inline gen/templates/data_source.go:155]: We can remove the comment for now. "id" is correct, though we would need to test import separately.

File: gen/templates/model.go [hunk 2]
```diff
@@ -135,6 +135,9 @@ func (data {{camelCase .Name}}) getPath() string {
 
 func (data {{camelCase .Name}}) toBody(ctx context.Context, state {{camelCase .Name}}) string {
 	body := ""
+	if data.Id.ValueString() != "" {
```
Comment (danischm) [inline gen/templates/model.go:138]: I think I have removed this in one of the recent commits to main, as we don't need to include the ID in the payload.

File: gen/templates/resource.go [hunk 3]
```diff
@@ -590,9 +590,9 @@ func (r *{{camelCase .Name}}Resource) ImportState(ctx context.Context, req resou
 	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("{{$attr.TfName}}"), idParts[{{$index}}])...)
 	{{- end}}
 	{{- end}}
-	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), idParts[{{subtract (importParts .Attributes) 1}}])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), idParts[{{subtract (importParts .Attributes) 1}}])...) // is this correct?
```
Comment (danischm) [inline gen/templates/resource.go:593]: same as above

File: gen/templates/resource.go [hunk 4]
```diff
@@ -590,9 +590,9 @@ func (r *{{camelCase .Name}}Resource) ImportState(ctx context.Context, req resou
...
 	{{- else}}
-	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp) // is this correct?
```
Comment (danischm) [inline gen/templates/resource.go:595]: same as above

File: openapiconverter/main.go [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/CiscoDevNet/terraform-provider-meraki/gen"
+	"github.com/tidwall/gjson"
+	"gopkg.in/yaml.v2"
```
Comment (danischm) [inline openapiconverter/main.go:12]: If possible let's use "v3" as this is what we are already using in the generator code.

**Review by danischm — CHANGES_REQUESTED** [review]

File: gen/templates/data_source.go (hunk 1 above)
Comment (mcparaf) [reply gen/templates/data_source.go]: removed

File: gen/templates/model.go (hunk 2 above)
Comment (mcparaf) [reply gen/templates/model.go]: removed

File: gen/templates/resource.go (hunk 3 above)
Comment (mcparaf) [reply gen/templates/resource.go]: removed

File: gen/templates/resource.go (hunk 4 above)
Comment (mcparaf) [reply gen/templates/resource.go]: removed

File: openapiconverter/main.go (hunk 5 above)
Comment (mcparaf) [reply openapiconverter/main.go]: changed to v3

File: networks_group_policy_manual.yaml [hunk 6]
```diff
@@ -0,0 +1,84 @@
+---
```
Comment (danischm) [inline networks_group_policy_manual.yaml:1]: This file can be removed I believe.

File: go.mod [hunk 7]
```diff
@@ -15,9 +16,12 @@ require (
 	github.com/tidwall/gjson v1.17.3
 	github.com/tidwall/sjson v1.2.5
 	golang.org/x/tools v0.25.0
+	gopkg.in/yaml.v2 v2.3.0
 	gopkg.in/yaml.v3 v3.0.1
 )
 
+replace github.com/netascode/go-meraki v0.0.0-20240901102824-a67592c39438 => /Users/maparafi/go-meraki
```
Comment (danischm) [inline go.mod:23]: This can be removed. I have already updated go.mod to point to the latest commit of go-meraki.

File: go.mod [hunk 8]
```diff
@@ -15,9 +16,12 @@ require (
 	github.com/tidwall/gjson v1.17.3
 	github.com/tidwall/sjson v1.2.5
 	golang.org/x/tools v0.25.0
+	gopkg.in/yaml.v2 v2.3.0
```
Comment (danischm) [inline go.mod:19]: This should be removed with "go mod tidy".

**Review by danischm — CHANGES_REQUESTED** [review]

**Pull request merged**